### PR TITLE
Add  VM basic tests, fix start/stop of execution component

### PR DESCRIPTION
**Pull request by gterzian**
Part of #1830

### CODE REVIEW

File: execution/src/vm.rs [hunk 1]
```diff
@@ -44,16 +44,16 @@ fn call(shared_env: &SharedExecutionContext, addr: Address, func_name: String, m
...
         (*exec_context_guard).call_stack.push_back(addr);
     }
 
     // run
     let mut run_failed = false;
-    match Instance::new(&target_module, &todo_imports) // TODO bring imports into the execution context (?)
-        .and_then(|inst| inst.exports.get_function(func_name))
-        .and_then(|f| f.native::<(), ()>()) // TODO figure out the "native" explicit parameters
-        .and_then(|f| f.call())
+    match Instance::new(&target_module, &ImportObject::new()) // TODO bring imports into the execution context (?)
```
Comment (gterzian) [inline execution/src/vm.rs:53]: Changes in this file are just quick changes to make it compile... @damip

**Comment by gterzian** [comment]
Ready for review, who wants to take a pass? @yvan-sraka @adrien-zinger @AureliaDolo @damip

File: execution/src/tests/scenarios_mandatories.rs [hunk 2]
```diff
@@ -5,6 +5,28 @@ use serial_test::serial;
...
+async fn test_execution_shutdown() {
+    let (_command_sender, _event_receiver, manager) = start_controller(ExecutionConfig {}, 2)
+        .await
+        .expect("Failed to start execution.");
+    manager.stop().await.expect("Failed to stop execution.");
+}
+
+#[tokio::test]
+#[serial]
+async fn test_sending_command() {
```
Comment (AureliaDolo) [inline execution/src/tests/scenarios_mandatories.rs:23]: Is there a way to auto generate tests that are just asserting that a command is propagated, not only here but every where else ?

Comment (adrien-zinger) [reply execution/src/tests/scenarios_mandatories.rs:23]: you mean like a macro?

**Review by adrien-zinger — COMMENTED** [review]
Look good to me

File: execution/src/vm.rs (hunk 1 above)
Comment (adrien-zinger) [reply execution/src/vm.rs:53]: Whatever, this part of the code will certainly change when we'll do the integration of the engine :+1:

File: execution/src/tests/scenarios_mandatories.rs (hunk 2 above)
Comment (gterzian) [reply execution/src/tests/scenarios_mandatories.rs:23]: I don't know, we could open an issue to discuss/investigate further.

**Comment by gterzian** [comment]
@damip Can you take a last look, the goal would be to merge this into your work at https://github.com/massalabs/massa/pull/1837.

File: execution/src/tests/scenarios_mandatories.rs (hunk 2 above)
Comment (adrien-zinger) [reply execution/src/tests/scenarios_mandatories.rs:23]: #1949

**Pull request merged**